src/leapYear.ts

```diff
@@ -0,0 +1,13 @@
+const leapYear = (inputYear: number): boolean => {
+  return isTypicalLeapYear(inputYear) || isAtypicalLeapYear(inputYear);
+};
+
+const isTypicalLeapYear = (inputYear: number): boolean => {
+  return inputYear % 4 === 0 && inputYear % 100 !== 0;
+};
+
+const isAtypicalLeapYear = (inputYear: number): boolean => {
+  return inputYear % 4 === 0 && inputYear % 100 === 0 && inputYear % 400 === 0;
```

Comment:
You dont need to check if its a multiple of 4 or 100 here
If its a multiple of 400, it's an atypical leap year